src/main/java/org/thoughtcrime/securesms/ProfileAdapter.java

```diff
@@ -0,0 +1,372 @@
+    boolean isGroup = dcChat != null && dcChat.getType() == DcChat.DC_CHAT_TYPE_GROUP;
+    boolean isSelfTalk = dcChat != null && dcChat.isSelfTalk();
+    boolean isDeviceTalk = dcChat != null && dcChat.isDeviceTalk();
+    memberCount = memberList!=null ? memberList.length : 0;
+
+    itemData.add(new ItemData(ITEM_AVATAR, null, 0));
+
+    if (isSelfTalk || dcContact != null && !dcContact.getStatus().isEmpty()) {
+      itemDataStatusText = isSelfTalk ? context.getString(R.string.saved_messages_explain) : dcContact.getStatus();
+      itemData.add(new ItemData(ITEM_SIGNATURE, itemDataStatusText, 0));
+    } else {
+      itemData.add(new ItemData(ITEM_DIVIDER, null, 0));
+    }
+
+    itemData.add(new ItemData(ITEM_ALL_MEDIA_BUTTON, context.getString(R.string.apps_and_media), R.drawable.ic_apps_24));
+
+    if (dcContact != null && !isDeviceTalk && !isSelfTalk) {
+      itemData.add(new ItemData(ITEM_SEND_MESSAGE_BUTTON, context.getString(R.string.send_message), R.drawable.ic_send_sms_white_24dp));
+    }
+
+    if (dcContact != null && !isDeviceTalk && !isSelfTalk) {
+      long lastSeenTimestamp = dcContact.getLastSeen();
+      String lastSeenTxt;
+      if (lastSeenTimestamp == 0) {
+        lastSeenTxt = context.getString(R.string.last_seen_unknown);
+      }
+      else {
+        lastSeenTxt = context.getString(R.string.last_seen_at, DateUtils.getExtendedTimeSpanString(context, lastSeenTimestamp));
+      }
+      itemData.add(new ItemData(ITEM_LAST_SEEN, lastSeenTxt, 0));
+    }
+
+    if (memberList!=null) {
+      itemData.add(new ItemData(ITEM_DIVIDER, null, 0));
+      if (dcChat != null) {
+        if (!isMailingList && dcChat.canSend()) {
+          itemData.add(new ItemData(ITEM_MEMBERS, DcContact.DC_CONTACT_ID_ADD_MEMBER, 0));
+          if (!isBroadcast) {
+            itemData.add(new ItemData(ITEM_MEMBERS, DcContact.DC_CONTACT_ID_QR_INVITE, 0));
+          }
+        }
+      }
+      for (int value : memberList) {
+        itemData.add(new ItemData(ITEM_MEMBERS, value, 0));
+      }
+    }
+
+    if (sharedChats != null && !isDeviceTalk) {
+      itemData.add(new ItemData(ITEM_HEADER, context.getString(R.string.profile_shared_chats), 0));
+      for (int i = 0; i < sharedChats.getCnt(); i++) {
+        itemData.add(new ItemData(ITEM_SHARED_CHATS, 0, i));
+      }
+    }
+
+    if (dcContact != null && !isDeviceTalk && !isSelfTalk) {
+      int verifierId = dcContact.getVerifierId();
+      if (verifierId != 0) {
+        String introducedBy;
+        if (verifierId == DcContact.DC_CONTACT_ID_SELF) {
+          introducedBy = context.getString(R.string.verified_by_you);
+        } else {
+          introducedBy = context.getString(R.string.verified_by, dcContext.getContact(verifierId).getDisplayName());
+        }
+        itemData.add(new ItemData(ITEM_INTRODUCED_BY, introducedBy, 0));
+      }
+
+      if (dcContact != null) {
+        itemData.add(new ItemData(ITEM_ADDRESS, dcContact.getAddr(), 0));
```

Comment:
BTW, for email-contacts, I'm not sure it's good to hide the address so much, but you probably discussed about this already, so it's fine not to discuss again.

Reply:
> maybe we can show it in the subtitle for email contact, that would match the pattern of repeating the chat-title enlarged

+1

> if so, maybe hide it alltogether otherwise? (it is still at encrypt info)

I think it would be kinda hard to figure where to look for this information, because it's not obvious to the user that you can get to the email address(es) by clicking on "Encryption"